Py3

JamesJeffryes opened the pull request:
Builds on PR #2 Merge that to simplify (a little)

eapearson commented:
I've looked at it locally, running the tests successfully, then using it locally against the jgi search ui. One small bug to fix, which I'll submit in a PR soon (tonight or in the morning) with some scripts and docs so I don't forget how to set it up as a local service. The jgi search project just sort of stopped, before the testing or docs could be completed, and before a number of jgi back end bugs could be fixed.

eapearson commented:
Hi @JamesJeffryes. So there were a few issues to resolve. 

One of the issues had to do with an out-of-date authclient. In the repo there are two copies -- the older broken one originally installed by the sdk in jgi_gateway and a newer working one in installed_clients. The sdk insists on using the one in jgi_gateway (e.g. when running kb-sdk compile it overwrites the import in the server file.) So I copied the one in installed_clients and it works fine.

Can the unused authclient in installed_clients be removed, or is something expecting it to be there? (not in the current codebase because I can see the answer is no, but e.g. in future sdk updates.)

Similarly, there are two copies of baseclient.py. The newer one and only one in use by this app is for the Workspace client in installed_clients. The older one is in jgi_gateway and unused. When installing the workspace client via the sdk (which I did just as a test), it does not bring in baseclient.py - it assumes that it is already present. However, the docs state that it should be located in the module's directory. It also seems that kb-sdk compile doesn't update that file either.

I'm wondering, generally, how these dependencies are kept up to date?
Is it fair to expect that this is something kb-sdk should handle, if not through kb-sdk compile perhaps through something like kb-sdk update or upgrade?

I'll have a PR based on this PR soon.

JamesJeffryes commented:
So, I thought that a those clients would be updated by kb-sdk compile but
they arn't if the python client is removed. I say update the the auth and
base clients in the jgi_gateway directory.

I think the auth client in installed_client is not used, it could be
removed. The sdk adds it basically because it's reusing code I think.

The baseclient is not used in this app (because I removed the python client
code) so it can be removed if you like.

Generally, the compile step refreshes these files and keeps them up to date
but not if the make file is changed to avoid generating clients.

On Wed, Mar 13, 2019 at 10:18 AM Erik Pearson <notifications@github.com>
wrote:

> Hi @JamesJeffryes <https://github.com/JamesJeffryes>. So there were a few
> issues to resolve.
>
> One of the issues had to do with an out-of-date authclient. In the repo
> there are two copies -- the older broken one originally installed by the
> sdk in jgi_gateway and a newer working one in installed_clients. The sdk
> insists on using the one in jgi_gateway (e.g. when running kb-sdk compile
> it overwrites the import in the server file.) So I copied the one in
> installed_clients and it works fine.
>
> Can the unused authclient in installed_clients be removed, or is something
> expecting it to be there? (not in the current codebase because I can see
> the answer is no, but e.g. in future sdk updates.)
>
> Similarly, there are two copies of baseclient.py. The newer one and only
> one in use by this app is for the Workspace client in installed_clients.
> The older one is in jgi_gateway and unused. When installing the workspace
> client via the sdk (which I did just as a test), it does not bring in
> baseclient.py - it assumes that it is already present. However, the docs
> state that it should be located in the module's directory. It also seems
> that kb-sdk compile doesn't update that file either.
>
> I'm wondering, generally, how these dependencies are kept up to date?
> Is it fair to expect that this is something kb-sdk should handle, if not
> through kb-sdk compile perhaps through something like kb-sdk update or
> upgrade?
>
> I'll have a PR based on this PR soon.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kbaseapps/jgi_gateway/pull/3#issuecomment-472467281>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIGfOdpkbQv7wtrpAC1fnKBrkuFdDDIgks5vWRbBgaJpZM4brmIF>
> .
>
